src/adapter/src/catalog/open.rs

```diff
@@ -676,14 +701,6 @@ impl Catalog {
                 txn.set_catalog_content_version(config.build_info.version.to_string())?;
             }
 
-            // Re-load the system configuration in case it changed after the migrations.
-            Catalog::load_system_configuration(
-                &mut state,
-                &mut txn,
-                &config.system_parameter_defaults,
-                config.remote_system_parameters.as_ref(),
-            )?;
```

Comment:
It's possible the system parameters change as part of a Catalog migration, which is why we reload the system config here. Depending on where we `apply(...)` updates it seems like this might get missed?

Reply:
I added a new migration (https://github.com/MaterializeInc/materialize/blob/5be93c9b31ad2ac9e870cebb1876775169194273/src/adapter/src/catalog/migrate.rs#L229-L237) that is called before any of the catalog state is loaded into memory (https://github.com/MaterializeInc/materialize/blob/5be93c9b31ad2ac9e870cebb1876775169194273/src/adapter/src/catalog/open.rs#L254-L256).

If we migrate the system parameters as part of that migration then there's no need to re-load them after migrating them. What are your thoughts on that? Does system config need a reference to a `CatalogState` to migrate or can we include them all in the earlier call that only has a reference to a `Transaction`?

Reply:
This looks great! It covers what I was concerned about